tests/integration-tests/tests/storage/test_fsx_lustre.py

```diff
@@ -214,6 +217,87 @@ def test_fsx_lustre_backup(region, pcluster_config_reader, clusters_factory, os,
 
 
+@pytest.mark.dimensions("us-west-2", "c5.xlarge", "alinux2", "slurm")
+@pytest.mark.usefixtures("instance")
+def test_existing_fsx(
+    region, existing_fsx, pcluster_config_reader, clusters_factory, os, scheduler, vpc_stack, test_datadir
+):
+    """
+    Test existing Fsx file system
+
+    Check fsx_fs_id provided in config file can be mounted correctly
+    """
+    mount_dir = "/fsx_mount_dir"
+    bucket_name = existing_fsx[1]
+    cluster_config = pcluster_config_reader(
+        bucket_name=bucket_name, mount_dir=mount_dir, fsx_fs_id=existing_fsx[0].cfn_resources["existingfsx"]
+    )
+    cluster = clusters_factory(cluster_config)
+
+    _test_fsx_lustre(cluster, region, scheduler, os, mount_dir, bucket_name, storage_type=None, auto_import_policy=None)
+
+
+@pytest.fixture()
+def existing_fsx(vpc_stack, cfn_stacks_factory, request, region, key_name, s3_bucket_factory, test_datadir):
+    # Create s3 bucket and upload test_file
+    bucket_name = s3_bucket_factory()
+    bucket = boto3.resource("s3", region_name=region).Bucket(bucket_name)
+    bucket.upload_file(str(test_datadir / "s3_test_file"), "s3_test_file")
+
+    # FSx stack
+    fsx_template = Template()
+    fsx_template.set_version()
+    fsx_template.set_description("Create FSx stack for testing existing FSx")
+
+    # Create security group. If using an existing file system
+    # It must be associated to a security group that allows inbound TCP traffic to port 988
+    existing_fsx_sg = ec2.SecurityGroup(
+        "ExistingFSxSecurityGroup",
+        GroupDescription="SecurityGroup for testing existing FSx",
+        SecurityGroupIngress=[
+            ec2.SecurityGroupRule(
+                IpProtocol="tcp",
+                FromPort="988",
+                ToPort="988",
+                CidrIp="0.0.0.0/0",
+            ),
+        ],
+        VpcId=vpc_stack.cfn_outputs["VpcId"],
+    )
+
+    fsx_filesystem = FileSystem(
+        title="existingfsx",
+        FileSystemType="LUSTRE",
+        StorageCapacity=1200,
+        SecurityGroupIds=[Ref(existing_fsx_sg)],
+        SubnetIds=[vpc_stack.cfn_outputs["PublicSubnetId"]],
+        LustreConfiguration=LustreConfiguration(
+            title="lustreConfiguration",
+            ImportPath="s3://{0}".format(bucket_name),
+            ExportPath="s3://{0}/export_dir".format(bucket_name),
+        ),
+        DependsOn=["ExistingFSxSecurityGroup"],
+    )
+    fsx_template.add_resource(existing_fsx_sg)
+    fsx_template.add_resource(fsx_filesystem)
+
+    fsx_stack = CfnStack(
+        name="integ-tests-fsx-{0}{1}{2}".format(
+            utils.random_alphanumeric(),
+            "-" if request.config.getoption("stackname_suffix") else "",
+            request.config.getoption("stackname_suffix"),
+        ),
+        region=region,
+        template=fsx_template.to_json(),
+    )
+    cfn_stacks_factory.create_stack(fsx_stack)
+
+    yield fsx_stack, bucket_name
```

Comment:
If we move the bucket creation outside of the fixture,  instead of returning the bucket_name we could eventually return the security_group that is a resource created within this fixture.

Reply:
Then instead of returning the fsx_stack we can instead return the fsx_id

Reply:
Done